### ISSUE TITLE
Tool hangs at disabling MDT options

### ISSUE THREAD

**Issue by Kevin-itdude** [maurice-daly/DriverAutomationTool#549]
Original hang was because I hadn't added the wmi class MS_SystemInformation to hardware inventory. Added all elements, got past that, now I have this:

You cannot call a method on a null-valued expression.
At line:1318 char:5
+ ...             $OSBuild = ($WindowsBuildHashTable.Item("$OSBuild")).Spli ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull
 
Cannot convert the "0BD2" value of type "System.String" to type "System.Collections.ArrayList".
At line:6047 char:6
+ ...             [System.Collections.ArrayList]$SearchList = $DellKnownBas ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [], RuntimeException
    + FullyQualifiedErrorId : ConvertToFinalInvalidCastException

What did I miss(configure)?

Thanks

**Comment by thomasforde** [comment]
i have a similar issue after upgrading to ConfigMgr 2403. the tool stops at "Disabling MDT Options" in the logs, and it wont detect any models or move past that. Are you using the same version?

**Comment by Kevin-itdude** [comment]
Yes, 2303 here.I’ve got it working now … What I found is that if it’s missing information in the configuration, it just waits, with no indication of why. The first couple times I was missing the scratch directory where it downloads the driver cabs and the directory where it places the driver packages when finished. I’ll have to look at the process log to confirm, but you may need to fill out the site server name and tell it to connect to move past that spot.Sent from my iPadOn Jun 19, 2024, at 4:00 AM, Thomas Førde ***@***.***> wrote:﻿
i have a similar issue after upgrading to ConfigMgr 2403. the tool stops at "Disabling MDT Options" in the logs, and it wont detect any models or move past that. Are you using the same version?

**Comment by Kevin-itdude** [comment]
Typo - On 2403 here.

**Comment by thomasforde** [comment]
Ye, its not working here.  Process logs says nothing more, all  the values are still there with my download paths,  site server settings, and package settings.  its connecting to configmgr server, and getting Dp's etc... but not finding any models which is probably the next step there. Winrm connection is established. Clicking find models and nothing happens, and no entry in the process log.
